### PR TITLE
add ⁇ ‼ unicode version

### PR DESCRIPTION
**Pull request by FCO**
add ⁇ ‼ unicode version

### CODE REVIEW

File: src/Perl6/Grammar.nqp [hunk 1]
```diff
@@ -1742,6 +1742,7 @@ grammar Perl6::Grammar is HLL::Grammar does STD {
     token term:sym<value>              { <value> }
     token term:sym<unquote>            { '{{{' <?{ $*IN_QUASI }> <statementlist> '}}}' }
     token term:sym<!!>                 { '!!' <?before \s> }  # actual error produced inside infix:<?? !!>
+    token term:sym<‼>                 { '‼' <?before \s> }  # actual error produced inside infix:<⁇ ‼>
```
Comment (AlexDaniel) [inline src/Perl6/Grammar.nqp]: Nitpicking, but why misaligned?

File: src/Perl6/Grammar.nqp [hunk 2]
```diff
@@ -4348,6 +4349,22 @@ grammar Perl6::Grammar is HLL::Grammar does STD {
         <O(|%conditional, :reducecheck<ternary>, :pasttype<if>)>
     }
 
+    token infix:sym<⁇ ‼> {
```
Comment (AlexDaniel) [inline src/Perl6/Grammar.nqp]: So this whole thing is just copy-pasted? Any better solution?

File: src/Perl6/Grammar.nqp (hunk 1 above)
Comment (FCO) [inline src/Perl6/Grammar.nqp]: I didn't see that... fixing...

Comment (AlexDaniel) [inline src/Perl6/Grammar.nqp]: Oh, ‼ is probably double-width in your font. OK.

File: src/Perl6/Grammar.nqp (hunk 2 above)
Comment (FCO) [inline src/Perl6/Grammar.nqp]: yes, just copy-pasted. I don't know if there is a better solution... any suggestion?

Comment (FCO) [inline src/Perl6/Grammar.nqp]: Not copy-pasted any more...

File: src/Perl6/Grammar.nqp [hunk 3]
```diff
@@ -4332,19 +4333,32 @@ grammar Perl6::Grammar is HLL::Grammar does STD {
     token infix:sym<min>  { <sym> >> <O(|%tight_or_minmax)> }
     token infix:sym<max>  { <sym> >> <O(|%tight_or_minmax)> }
 
+    token ternary-explain {
+        || <?before '::' <.-[=]>> { self.typed_panic: "X::Syntax::ConditionalOperator::SecondPartInvalid", second-part => "::" }
```
Comment (samcv) [inline src/Perl6/Grammar.nqp]: I don't think this should have the leading `||`

Comment (AlexDaniel) [inline src/Perl6/Grammar.nqp]: Why not? That's being used almost everywhere, and it's just a syntactic thing to make the stuff aligned.

Comment (samcv) [inline src/Perl6/Grammar.nqp]: There's no reason for it to be there. Change it to spaces or something. Though are you saying we have useless leading `||` in other places too?

Comment (AlexDaniel) [inline src/Perl6/Grammar.nqp]: Yes. In fact, the syntax itself was added on purpose.

Don't change it.

**Comment by zoffixznet** [comment]
@FCO, did you run `TEST_JOBS=8 make spectest` (or better yet, `TEST_JOBS=8 make stresstest`, since this is a pretty core change)?

See you moved some stuff to a separate token, but it still uses `~$<EXPR>` which isn't available to it anymore, so it's always Nil. If our tests don't catch that, we need to improve them.

**Comment by FCO** [comment]
I made the ?? !! and the ⁇ ‼ a single token. I think that fixes the @zoffixznet, @AlexDaniel and @samcv  issues. I've ran the spec tests and the stress tests, I got 2 errors on the first and 1 on the last one. Both on spec tests worked when I ran only the test (t/spec/S10-packages/basic.rakudo.moar and t/spec/S17-supply/watch-path.t). The stresstest error was on file t/spec/S10-packages/precompilation.rakudo.moar 2 errors on TODOs and 1 real error. @zoffixznet told me on irc that that test "flop on occasion".

**Comment by AlexDaniel** [comment]
Looks nicer indeed.

**Comment by zoffixznet** [comment]
Thanks.

File: src/Perl6/Grammar.nqp [hunk 4]
```diff
@@ -4391,17 +4392,17 @@ grammar Perl6::Grammar is HLL::Grammar does STD {
     token infix:sym<max>  { <sym> >> <O(|%tight_or_minmax)> }
 
     token infix:sym<?? !!> {
-        :my $*GOAL := '!!';
-        $<sym>='??'
+        $<sym>=['⁇' || '??']
+        {}:my $*GOAL := $<sym> eq '??' ?? '!!' !! '‼';
```
Comment (vendethiel) [inline src/Perl6/Grammar.nqp]: why the leading `{}`?

Comment (zoffixznet) [inline src/Perl6/Grammar.nqp]: To update $/

**Pull request merged**